### PR TITLE
`none` algorithme and fixed signature for `all` and `any`

### PR DESCRIPTION
**Pull request by jonathanpoelen**

### CODE REVIEW

File: src/kvasir/mpl/algorithm/none.hpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+//          Copyright Odin Holmes 2016.
```
Comment (viboes) [inline src/kvasir/mpl/algorithm/none.hpp:1]: Copyright?

File: test/algorithm/none.hpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+#include <kvasir/mpl/algorithm/none.hpp>
+#include <kvasir/mpl/functional/bind.hpp>
+#include <kvasir/mpl/types/list.hpp>
+
+namespace {
+	template <typename T>
+	using comp = std::is_same<int, T>;
+
+	using namespace kvasir;
+	static_assert(!mpl::eager::none<mpl::list<void, char, int, float>, comp>::value, "");
```
Comment (viboes) [inline test/algorithm/none.hpp:18]: Add test with empty list
